Makefile

```diff
@@ -197,41 +226,63 @@ check_cert_manager:
                fi;\
        fi
 
+# Location to install dependencies
+LOCALBIN ?= $(shell pwd)/bin
+.PHONY: local-bin
+local-bin:
+	mkdir -p $(LOCALBIN)
+
+# Location to install test dependencies
+TESTBIN ?= $(shell pwd)/testbin
+.PHONY: test-bin
+test-bin:
+	mkdir -p $(TESTBIN)
+
 # Download controller-gen locally if necessary
-CONTROLLER_GEN = $(shell pwd)/bin/controller-gen
-controller-gen:
-	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1)
+CONTROLLER_GEN = $(LOCALBIN)/controller-gen
+.PHONY: controller-gen
+controller-gen: local-bin
+	[ -f $(CONTROLLER_GEN) ] || GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-tools/cmd/controller-gen@v$(CONTROLLER_GEN_VERSION)
 
 # Download kustomize locally if necessary
-KUSTOMIZE = $(shell pwd)/bin/kustomize
-kustomize:
-	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v3@v$(KUSTOMIZE_VERSION))
+KUSTOMIZE_INSTALL_SCRIPT ?= "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"
+KUSTOMIZE = $(LOCALBIN)/kustomize
+.PHONY: kustomize
+kustomize: local-bin
+	[ -f $(KUSTOMIZE) ] || { curl -s $(KUSTOMIZE_INSTALL_SCRIPT) | bash -s -- $(subst v,,$(KUSTOMIZE_VERSION)) $(LOCALBIN); }
 
 # Download addlicense locally if necessary
-ADDLICENSE = $(shell pwd)/bin/addlicense
-addlicense:
-	$(call go-get-tool,$(ADDLICENSE),github.com/google/addlicense@v$(ADDLICENSE_VERSION))
-
-# go-get-tool will 'go get' any package $2 and install it to $1.
-PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
-define go-get-tool
-@[ -f $(1) ] || { \
-set -e ;\
-TMP_DIR=$$(mktemp -d) ;\
-cd $$TMP_DIR ;\
-go mod init tmp ;\
-echo "Downloading $(2)" ;\
-GOBIN=$(PROJECT_DIR)/bin go get $(2) ;\
-rm -rf $$TMP_DIR ;\
-}
-endef
+ADDLICENSE = $(LOCALBIN)/addlicense
+.PHONY: addlicense
+addlicense: local-bin
+	[ -f $(ADDLICENSE) ] || GOBIN=$(LOCALBIN) go install github.com/google/addlicense@v$(ADDLICENSE_VERSION)
+
+# Download setup-envtest locally if necessary
+SETUP_ENVTEST = $(TESTBIN)/setup-envtest
```

Comment:
I noticed that the documentation uses `ENVTEST` for this variable name. Should we use that to minimize differences?

Reply:
Oh yes suree I will change this! Originally I used `SETUP_ENVTEST` to match its binary name :D